File: backend/src/main/java/com/dreamCollection/place/entity/Place.java
package com.dreamCollection.place.entity;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Entity
@Table(
        name = "place",
        uniqueConstraints = @UniqueConstraint(
                name = "uk_place_source_external",
                columnNames = {"source", "external_place_id"}
        )
)
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@AllArgsConstructor
@Builder
public class Place {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "external_place_id", nullable = false)
    private String externalPlaceId;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 30)
    private PlaceSource source;

    @Column(nullable = false, length = 200)
    private String name;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 30)
    private PlaceCategory category;

    @Column(length = 500)
    private String address;

    @Column(precision = 10, scale = 7)
    private BigDecimal latitude;

    @Column(precision = 10, scale = 7)
    private BigDecimal longitude;

    @Column(nullable = false, length = 100)
    private String city;

    @Column(length = 100)
    private String region;

    @Column(length = 100)
    private String country;

    @Column(name = "country_code", length = 2)
    private String countryCode;

    @Column(precision = 3, scale = 2)
    private BigDecimal rating;

    @Column(name = "review_count")
    private Integer reviewCount;

    @Column(name = "price_level", length = 20)
    private String priceLevel;

    @Column(name = "place_type", length = 100)
    private String placeType;

    @Column(name = "opening_hours_text", length = 500)
    private String openingHoursText;

    @Column(columnDefinition = "TEXT")
    private String description;

    @Column(name = "image_url", length = 1000)
    private String imageUrl;

    @Column(name = "external_url", length = 1000)
    private String externalUrl;

    @Builder.Default
    @Column(nullable = false)
    private boolean active = true;

    @CreationTimestamp
    @Column(nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @UpdateTimestamp
    @Column(nullable = false)
    private LocalDateTime updatedAt;
}

File: frontend/src/trip/api/place.ts
import apiClient from "@/common/api/client";
import type { ApiResponse } from "@/types";

export type PlaceCategory =
  | "ATTRACTION"
  | "RESTAURANT"
  | "CAFE"
  | "SHOPPING"
  | "NATURE"
  | "CULTURE"
  | "ACTIVITY"
  | "TRANSPORT"
  | "HOTEL";

export interface PlaceOption {
  id: number;
  externalPlaceId: string;
  source: string;
  name: string;
  category: PlaceCategory;
  address?: string | null;
  latitude?: number | null;
  longitude?: number | null;
  city?: string | null;
  region?: string | null;
  country?: string | null;
  countryCode?: string | null;
  rating?: number | null;
  reviewCount?: number | null;
  priceLevel?: string | null;
  placeType?: string | null;
  openingHoursText?: string | null;
  description?: string | null;
  imageUrl?: string | null;
  externalUrl?: string | null;
}

export const placeApi = {
  getPlaces: async (city: string, category?: PlaceCategory) => {
    const response = await apiClient.get<ApiResponse<PlaceOption[]>>(
      "/places",
      {
        params: {
          city,
          ...(category ? { category } : {}),
        },
      },
    );

    return response.data.data;
  },
};
